### PR TITLE
Crds

### PR DESCRIPTION
**Pull request by shtripat**

### CODE REVIEW

File: pkg/apis/operator/v1alpha1/glustercluster_types.go [hunk 1]
```diff
@@ -4,19 +4,62 @@ import (
...
+	StorageClassName string `json:"storageClassName"`
+	Capacity         string `json:"capacity"`
+}
+
+// GlusterNodeTemplate defines a gluster node's template
+type GlusterNodeTemplate struct {
+	Name       string                    `json:"name"`
+	Zone       string                    `json:"zone"`
+	Thresholds GlusterNodeThreshold      `json:"thresholds"`
+	Affinity   string                    `json:"nodeAffinity"`
```
Comment (JohnStrunk) [inline pkg/apis/operator/v1alpha1/glustercluster_types.go]: I think this type should be imported from kube. It should be this structure: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#affinity-v1-core

File: pkg/apis/operator/v1alpha1/glustercluster_types.go [hunk 2]
```diff
@@ -4,19 +4,62 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// GlusterClusterCredentials defines secret credentials
+type GlusterClusterCredentials struct {
```
Comment (JohnStrunk) [inline pkg/apis/operator/v1alpha1/glustercluster_types.go]: Could this and `Credentials` in `glusternode_types.go` be named similarly?

**Review by JohnStrunk — COMMENTED** [review]
Nice start. Thanks!

File: pkg/apis/operator/v1alpha1/glustercluster_types.go (hunk 1 above)
Comment (shtripat) [reply pkg/apis/operator/v1alpha1/glustercluster_types.go]: Ack

File: pkg/apis/operator/v1alpha1/glustercluster_types.go (hunk 2 above)
Comment (shtripat) [reply pkg/apis/operator/v1alpha1/glustercluster_types.go]: Ack

**Pull request merged**